### ISSUE TITLE
Prevent specifying error type in `t.notThrows()`

### ISSUE THREAD

**Issue by sindresorhus** [avajs/eslint-plugin-ava#178]
[<img alt="Issuehunt badges" src="https://img.shields.io/badge/IssueHunt-%2460%20Rewarded-%237E24E3.svg" />](https://oss.issuehunt.io/r/avajs/eslint-plugin-ava/issues/178)

It's an easy mistake to make if you have many `t.throws()` statements and then some `t.notThrows()` statements.

For example: https://github.com/sindresorhus/p-queue/commit/80e8e25926a6632d0429ebb526ee78a68f01d565

```js
// Fail
t.notThrows(() => {
 t.pass();
}, TypeError);

// Ok
t.notThrows(() => {
 t.pass();
});
```

It should fail if it's a variable matching this regex: `/^(?:[A-Z][a-z0-9]*)*Error$/`

---

<details>
<summary>
<b>IssueHunt Summary</b>
</summary>

### Backers (Total: $60.00)

- [<img src='https://avatars3.githubusercontent.com/u/44827199?v=4' alt='issuehunt' width=24 height=24> issuehunt](https://oss.issuehunt.io/u/issuehunt) ($60.00)

### Submitted pull Requests
- [#240 rule `prevent-errortype`](https://oss.issuehunt.io/r/avajs/eslint-plugin-ava/pull/240)
- [#326 Add `no-error-ctor-with-notthrows` rule](https://oss.issuehunt.io/r/avajs/eslint-plugin-ava/pull/326)
- [#835 Add `request` and `options` to `beforeError` hook state](https://oss.issuehunt.io/r/sindresorhus/ky/pull/835)
---

### Tips

- Checkout the [Issuehunt explorer](https://oss.issuehunt.io/r/avajs/eslint-plugin-ava/) to discover more funded issues.
- Need some help from other developers? [Add your repositories](https://oss.issuehunt.io/r/new) on IssueHunt to raise funds.
</details>

**Comment by sindresorhus** [comment]
@novemberborn In AVA, I think we should assert it's a string or undefined at runtime. What do you think?

**Comment by novemberborn** [comment]
@sindresorhus https://github.com/avajs/ava/issues/1125

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $60.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/49722492/issues/178) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/49722492/issues/178).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by gurungrahul2** [comment]
@sindresorhus @novemberborn I tried to give it a shot please review and guide further. #240
